nibabel/streamlines/tck.py

```diff
@@ -401,46 +396,36 @@ def _read(cls, fileobj, header, buffer_size=4):
             f.seek(header["_offset_data"], os.SEEK_SET)
 
             eof = False
-            buffs = []
             n_streams = 0
-
+            leftover = np.empty([0,3])
```

Comment:
This should be the same `dtype` as `coords`, below, otherwise things will get upcast to `float64`:

```suggestion
            leftover = np.empty((0, 3), dtype='<f4')
```